File: policyliner-demo-ui/src/app/dtos/pagedResponse.dto.ts
export class PagedResponseDTO<T> {
  elements: T[] = [];
  totalElements: number = 0;
  totalPages: number = 0;
  currentPage: number = 0;
  pageSize: number = 0;

  constructor(obj: any) {
    if (obj){
      this.elements = obj.elements;
      this.totalElements = obj.totalElements;
      this.totalPages = obj.totalPages;
      this.currentPage = obj.currentPage;
      this.pageSize = obj.pageSize;
    }

  }
}

File: policyliner/src/main/java/de/tub/dima/policyliner/dto/PagedResponseDTO.java
package de.tub.dima.policyliner.dto;

import java.util.List;

public class PagedResponseDTO<T> {
    private List<T> elements;
    private Long totalElements;
    private Integer totalPages;
    private Integer currentPage;
    private Integer pageSize;

    public PagedResponseDTO(List<T> elements, Long totalElements, Integer totalPages, Integer currentPage, Integer pageSize) {
        this.elements = elements;
        this.totalElements = totalElements;
        this.totalPages = totalPages;
        this.currentPage = currentPage;
        this.pageSize = pageSize;
    }

    public PagedResponseDTO() {
    }

    public List<T> getElements() {
        return elements;
    }

    public void setElements(List<T> elements) {
        this.elements = elements;
    }

    public Long getTotalElements() {
        return totalElements;
    }

    public void setTotalElements(Long totalElements) {
        this.totalElements = totalElements;
    }

    public Integer getTotalPages() {
        return totalPages;
    }

    public void setTotalPages(Integer totalPages) {
        this.totalPages = totalPages;
    }

    public Integer getCurrentPage() {
        return currentPage;
    }

    public void setCurrentPage(Integer currentPage) {
        this.currentPage = currentPage;
    }

    public Integer getPageSize() {
        return pageSize;
    }

    public void setPageSize(Integer pageSize) {
        this.pageSize = pageSize;
    }
}
